### PR TITLE
refactor: unify policy contract, modularize hotspots, remove duplicated infra logic

### LINKED ISSUE #55: Platform hardening/refactor: unify policy contract, modularize hotspots, and remove duplicated infra logic
The codebase has accumulated several related maintainability risks that are now slowing safe changes and increasing regression risk:

•  Duplicate gateway policy schema exists in both operator and proxy implementations, which can drift.
•  Very large files (internal/cli/setup.go, internal/operator/controller.go, services/mcp-proxy/main.go) mix too many concerns.
•  Setup manifest mutation relies on regex/string replacement, which is brittle.
•  Shared helper logic (envOr, boolEnv, otlpTraceOptions, auth parsing helpers) is duplicated across services.
•  Trust/policy enum vocabulary is inconsistent between API and shared packages.
•  Runtime API action endpoints use manual string path parsing.
•  Status condition updates in the operator are repetitive and error-prone.
•  Contract-level compatibility testing is missing between operator-rendered policy and proxy-consumed policy.

Proposed scope
•  Introduce a single shared gateway policy package used by operator + proxy.
•  Refactor setup/controller/proxy hotspots into smaller modules with clear ownership boundaries.
•  Replace regex-based YAML edits with structured manifest mutation/patching.
•  Centralize shared service helpers into common packages.
•  Normalize trust model definitions (or add explicit, tested translation boundaries).
•  Replace manual runtime path parsing with explicit route param handling.
•  Refactor status condition writing into reusable helper(s).
•  Add contract tests proving operator policy output remains consumable by proxy.

Acceptance criteria
•  No duplicate gateway policy struct definitions remain in operator/proxy.
•  Manifest mutation is structured (not regex/string patching in core setup flow).
•  Core hotspot files are split with reduced responsibility overlap.
•  Duplicated helper logic is consolidated and consumed by API/proxy/other services.
•  Trust-level semantics are consistent and covered by tests.
•  Runtime action routes use robust parameter handling with consistent method validation.
•  Operator status condition updates are generated through a shared path.
•  New/updated tests cover policy compatibility and critical refactored paths.

### PR DESCRIPTION
**Pull request by Agent-Hellboy**
…ed infra logic

This change addresses issue #55 by:

New Shared Packages:
- pkg/policy/ - Gateway policy types shared between operator and proxy
- pkg/serviceutil/ - Service helpers (env, otel, auth, http, routing)
- pkg/manifest/ - Structured YAML mutation (replaces regex-based approach)
- pkg/operatorutil/ - Operator status condition helpers

Key Changes:
- internal/operator/controller.go: Uses shared policy types, improved status helper
- internal/cli/setup.go: Uses structured manifest mutation instead of regex
- services/mcp-proxy/main.go: Uses shared policy types and service utilities
- services/api/main.go & runtime.go: Uses shared service utilities and routing helpers

Benefits:
- No duplicate gateway policy struct definitions
- Structured YAML mutation eliminates regex injection risks
- Centralized helper logic reduces drift
- Trust-level semantics consistent across services
- Robust route parameter handling with validation
- Contract tests ensure operator-proxy compatibility

fixes #55 

## Summary by CodeRabbit

* **New Features**
  * Added OpenTelemetry OTLP HTTP trace exporter support and new standardized gateway policy document types.

* **Improvements**
  * Structured manifest mutation for operator manifests and improved operator status/condition management.
  * Shared service utilities for auth, routing, env, JSON responses, and OTLP option construction.
  * Trust-level normalization and policy helper utilities.

* **Tests**
  * Added comprehensive unit tests for manifest mutator, policy, and service utilities.

* **Chores**
  * Module dependency upgrades and module import path tidy-up.

### CODE REVIEW

File: pkg/operatorutil/status.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		if apierrors.IsConflict(err) {
+			logger.V(1).Info("Status update conflict (expected in concurrent reconciles), will retry on next reconcile", "key", key)
+		} else {
+			logger.Error(err, "Failed to update status", "key", key)
+		}
+		return err
+	}
+
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (Agent-Hellboy) [reply pkg/operatorutil/status.go:100]: @coderabbitai, why are we touching user things as a platform, trace patch where we are touching the user-level namespace
